iroh-net/src/hp/derp/http/client.rs

```diff
@@ -832,6 +867,125 @@ impl Client {
+            }
+
+            let peers_present = PeersPresent::new(server_public_key.clone());
+            tracing::info!("Connected to mesh derp server {server_public_key:?}");
+
+            // receive detail loop
+            loop {
+                let (msg, conn_gen) = match self.recv_detail().await {
+                    Ok(res) => res,
+                    Err(e) => {
+                        tracing::warn!("recv error: {e:?}");
+                        tokio::time::sleep(redial_delay).await;
+                        break;
+                    }
+                };
+                if conn_gen != last_conn_gen {
+                    // TODO: In the current set up, once we know that this current
+                    // connection is a new connection, we must re-establish to the
+                    // server that we want to watch connection changes.
+                    // In the future, depending on how we handle multiple connections
+                    // for the same peer key in the derp server in the future, we may
+                    // be okay to just listen for future peer present
+                    // messages without re establishing this connection as a "watcher"
+                    break;
+                }
+                match msg {
+                    ReceivedMessage::PeerPresent(key) => {
+                        // ignore notifications about ourself
+                        if key == ignore_key {
+                            continue;
+                        }
+                        peers_present.insert(key.clone()).await;
+                        packet_forwarder_handler.add_packet_forwarder(key, self.clone())?;
+                    }
+                    ReceivedMessage::PeerGone(key) => {
+                        peers_present.remove(&key).await;
+                        packet_forwarder_handler.remove_packet_forwarder(key)?;
+                    }
+                    _ => {}
+                }
+            }
+        }
+    }
+}
+
+const PEERS_PRESENT_LOGGING_DELAY: Duration = Duration::from_secs(5);
+const PEERS_PRESENT_LOGGING_INTERVAL: Duration = Duration::from_secs(10);
+
+#[derive(Debug)]
+/// A struct to track and log the peers available on the remote server
+struct PeersPresent {
+    /// map of keys of the peers that can be contacted on the given remote server
+    map: Arc<RwLock<HashSet<key::node::PublicKey>>>,
+    /// periodic logging task
+    logging_task: JoinHandle<()>,
+}
+
+impl PeersPresent {
+    fn new(remote_server_key: key::node::PublicKey) -> Self {
+        let map = Arc::new(RwLock::new(HashSet::default()));
+        let log_map = Arc::clone(&map);
+        let logging_task = tokio::spawn(async move {
+            let start = Instant::now() + PEERS_PRESENT_LOGGING_DELAY;
+            let mut status_logging_interval =
+                tokio::time::interval_at(start, PEERS_PRESENT_LOGGING_INTERVAL);
+
+            loop {
+                status_logging_interval.tick().await;
```

Comment:
nit: might be nicer to use [IntervalStream](https://docs.rs/tokio-stream/latest/tokio_stream/wrappers/struct.IntervalStream.html) and do sth like 

```rust
while status_logging_interval.next().await.is_some() {
  // .. 
}
```
(don't know if I like it better, just a thought..)